test/unit/gateways/worldpay_test.rb

```diff
@@ -213,6 +281,35 @@ def test_successful_purchase
     assert_success response
   end
 
+  def test_transaction_with_level_two_data
+    options = @options.merge(@level_two_data)
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card, options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<invoiceReferenceNumber>INV12233565</invoiceReferenceNumber>), data
+      assert_match %r(<customerReference>CUST00000101</customerReference>), data
+      assert_match %r(<cardAcceptorTaxId>VAT1999292</cardAcceptorTaxId>), data
+      assert_match %r(<salesTax><amountvalue="20"currencyCode="USD"exponent="2"/></salesTax>), data.gsub(/\s+/, '')
+    end.respond_with(successful_authorize_response)
+    assert_success response
+  end
+
+  def test_transaction_with_level_three_data
+    options = @options.merge(@level_three_data)
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card, options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<customerReference>CUST00000102</customerReference>), data
+      assert_match %r(<cardAcceptorTaxId>VAT1999285</cardAcceptorTaxId>), data
+      assert_match %r(<salesTax><amountvalue="20"currencyCode="USD"exponent="2"/></salesTax>), data.gsub(/\s+/, '')
+      assert_match %r(<discountAmount><amountvalue="1"currencyCode="USD"exponent="2"/></discountAmount>), data.gsub(/\s+/, '')
+      assert_match %r(<shippingAmount><amountvalue="50"currencyCode="USD"exponent="2"/></shippingAmount>), data.gsub(/\s+/, '')
+      assert_match %r(<dutyAmount><amountvalue="20"currencyCode="USD"exponent="2"/></dutyAmount>), data.gsub(/\s+/, '')
+      assert_match %r(<item><description>Laptop14</description><productCode>LP00125</productCode><commodityCode>COM00125</commodityCode><quantity>2</quantity><unitCost><amountvalue="1500"currencyCode="USD"exponent="2"/></unitCost><unitOfMeasure>each</unitOfMeasure><itemTotal><amountvalue="3000"currencyCode="USD"exponent="2"/></itemTotal><itemTotalWithTax><amountvalue="3500"currencyCode="USD"exponent="2"/></itemTotalWithTax><itemDiscountAmount><amountvalue="200"currencyCode="USD"exponent="2"/></itemDiscountAmount><taxAmount><amountvalue="500"currencyCode="USD"exponent="2"/></taxAmount></item>), data.gsub(/\s+/, '')
+    end.respond_with(successful_authorize_response)
+    assert_success response
+  end
+
```

Comment:
💬  Comment:

Just wondering if the customerReference  or cardAcceptorTaxId are considered personal sensitive data ?

Reply:
mm good point, but I don't thing